docs/development/malware-checks.rst

```diff
@@ -0,0 +1,109 @@
+
+|VerdictLifecycle|
+
+Checks can be triggered in the following ways:
+
+* A PyPI user uploads a new File, Release or Project;
+* A schedule;
+* A PyPI administrator initiates an evaluation run.
+
+All of the above triggers call the `IMalwareCheckService`_ factory to determine how to execute the check. On production, 
+the `DatabaseMalwareCheckService`_ is returned, which runs the check and produces one or more verdicts. PyPI administrator and moderators are continuously reviewing the verdicts in the warehouse admin, making determinations about the accuracy of checks, about taking further action if needed (e.g. removing a malicious package surfaced by a verdict).
+
+Contributing
+------------
+
+Check Lifecycle
+~~~~~~~~~~~~~~~
+|CheckLifecycle|
+
+Ideas for new malware checks should first be shared by `opening an issue`_. This will initiate a discussion with PyPI administrators and among the broader Python community about the impact of the proposed check. After soliciting feedback, `open a pull request to master`_ containing the code for the new check, unit tests, and accompanying documentation. Once the code is reviewed and merged, it will automatically be deployed to production. PyPI administrators can begin evaluating the malware check by moving it into the `evaluation` state in the check admin and triggering an evaluation run.
+
+The evaluation run generates verdicts, which are viewable in the verdicts admin. After reviewing the verdicts, the administrator will make a determination and communicate it to the check developer in the initial issue. Here are the possible outcomes:
+
+* The check provides a low-quality or noisy signal (e.g. many false positives), and should be removed. At this point, the check will be moved into a `wiped_out` state, removing all verdicts generated by the evaluation run, and the code for the check will be removed in the next release.
+* The check provides a useful signal, but requires modifications. The administrator will request changes in the initial issue.
+* The check provides a useful signal, and the administrator enables it.
+
+Adding New Checks
+~~~~~~~~~~~~~~~~~
+All active checks are defined as classes in the `warehouse/malware/checks/`_ directory, and exported from ``__init__.py``. The checks in `tests/common/checks/`_ can serve as templates for developing new checks. Simply copy/paste the desired check template into `warehouse/malware/checks/`_ and edit the `dunder init file`_ to get started.
+
+All malware check classes should inherit from ``warehouse.malware.checks.base.MalwareCheckBase``, define a `scan` method, and set the following fields as class attributes:
+
+* ``version`` - ``1`` for new checks, incrementing by one with every subsequent change
+* ``short_description`` - a terse description of the check's purpose
+* ``long_description`` - a more detailed rationale for the check
+* ``check_type`` - either ``event_hook`` or ``scheduled``
```

Comment:
Nit: maybe add quotes to `event_hook` and `scheduled` here to indicate that they're string literals?